### ISSUE TITLE
[A11y] Theme Dynamic News Missing Label for Header Search Button

### ISSUE THREAD

**Issue by masperber** [Automattic/wp-calypso#64524]
### Quick summary

In the Dynamic News theme, the Header Search Button is missing a label. Though this theme is retired, this bug should still be fixed for accessibility reasons.

### Steps to reproduce

1. Install Dynamic News on a site.
2. Enable Header Search from Customize > Theme
3. Inspect element on the Search button.

### What you expected to happen

The element

```
<button type="submit" class="search-submit">
```

Should have a <label> with screen reader text.

### What actually happened

The <button> element lacks a <label>.

### Context

Customer report in 35767098-hc

Please follow up with the customer in 5287019-zen when resolved.

### Simple, Atomic or both?

Simple, Atomic

### Theme-specific issue?

Dynamic News

### Reproducibility

Consistent

### Severity

All
